### PR TITLE
Xe: Remove unused emitXeHardwareMask option

### PR DESCRIPTION
**Pull request by e3m3**
Changes:
*   Remove `--emit-xe-hardware-mask` flag.
    Remove `emitXeHardwareMask` code paths.
    See [discussion](https://github.com/ispc/ispc/pull/2494#discussion_r1267393765).
    | `src/ctx.cpp`,  `src/ctx.h`,
    | `src/expr.cpp`,
    | `src/func.cpp`,
    | `src/ispc.cpp`, `src/ispc.h`,
    | `src/main.cpp`,
    | `src/stmt.cpp`, `src/stmt.h`

### CODE REVIEW

**Comment by aneshlya** [comment]
It's not blocking for #2494. I need to do several experiments with VC BE before reviewing this PR to ensure that these code paths can be safely removed and they will not be needed in the future.

File: src/ctx.cpp [hunk 1]
```diff
@@ -3443,14 +3258,7 @@ llvm::Value *FunctionEmitContext::CallInst(llvm::Value *func, const FunctionType
...
         AddressInfo *maskPtrInfo = AllocaInst(LLVMTypes::MaskType);
-        if (emitXeHardwareMask()) {
-#ifdef ISPC_XE_ENABLED
-            // Current mask will be calculated according to EM mask
-            oldFullMask = XeSimdCFPredicate(LLVMMaskAllOn);
-#endif
-        } else {
-            oldFullMask = GetFullMask();
-        }
+        oldFullMask = GetFullMask();
```
Comment (nurmukhametov) [inline src/ctx.cpp:3261]:
```c++
llvm::Value *oldFullMask = GetFullMask();
```

File: src/stmt.cpp [hunk 2]
```diff
@@ -2159,15 +1954,7 @@ void ForeachActiveStmt::EmitCode(FunctionEmitContext *ctx) const {
...
-#ifdef ISPC_XE_ENABLED
-    if (ctx->emitXeHardwareMask()) {
-        // Emulate uniform to make proper continue handler
-        uniformEmulated = true;
-        // Current mask will be calculated according to EM mask
-        oldFullMask = ctx->XeSimdCFPredicate(LLVMMaskAllOn);
-    } else
-#endif
-        oldFullMask = ctx->GetFullMask();
+    oldFullMask = ctx->GetFullMask();
```
Comment (nurmukhametov) [inline src/stmt.cpp:1957]: same here

**Comment by e3m3** [comment]
>

**Comment by e3m3** [comment]
> It's not blocking for #2494. I need to do several experiments with VC BE before reviewing this PR to ensure that these code paths can be safely removed and they will not be needed in the future.

Ok - I will leave the unused code path in #2494. For the comments regarding the EmitCodeForXe, I will assume it's okay to merge them since they don't affect our workloads/tests.

**Comment by e3m3** [comment]
We will revisit this PR when VC BE SimdCF changes land. Closing.

**Pull request closed without merging**